### PR TITLE
fix: don't fail when MemberStatus doesn't have any condition

### PR DESCRIPTION
**Pull request by MatousJobanek**
When the MemberStatus is just created, it doesn't have any condition. The synchronization from host can panic with:
```
{"level":"error","ts":1624435095.8653722,"logger":"controllers.ToolchainStatus","msg":"the memberstatus ready condition is not true","Request.Namespace":"toolchain-host-operator","Request.Name":"toolchain-status","error":"member cluster member-xcoulon-20210623a.devcluster.openshift.com not ready","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/Users/xcoulon/code/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\ngithub.com/codeready-toolchain/host-operator/controllers/toolchainstatus.(*Reconciler).membersHandleStatus\n\t/Users/xcoulon/code/go/src/github.com/codeready-toolchain/host-operator/controllers/toolchainstatus/toolchainstatus_controller.go:351\ngithub.com/codeready-toolchain/host-operator/controllers/toolchainstatus.(*Reconciler).aggregateAndUpdateStatus\n\t/Users/xcoulon/code/go/src/github.com/codeready-toolchain/host-operator/controllers/toolchainstatus/toolchainstatus_controller.go:176\ngithub.com/codeready-toolchain/host-operator/controllers/toolchainstatus.(*Reconciler).Reconcile\n\t/Users/xcoulon/code/go/src/github.com/codeready-toolchain/host-operator/controllers/toolchainstatus/toolchainstatus_controller.go:136\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/Users/xcoulon/code/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.0/pkg/internal/controller/controller.go:256\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/Users/xcoulon/code/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.0/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/Users/xcoulon/code/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.0/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/Users/xcoulon/code/go/pkg/mod/k8s.io/apimachinery@v0.18.3/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/Users/xcoulon/code/go/pkg/mod/k8s.io/apimachinery@v0.18.3/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/Users/xcoulon/code/go/pkg/mod/k8s.io/apimachinery@v0.18.3/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/Users/xcoulon/code/go/pkg/mod/k8s.io/apimachinery@v0.18.3/pkg/util/wait/wait.go:90"}
E0623 07:58:15.865506       1 runtime.go:78] Observed a panic: runtime.boundsError{x:0, y:0, signed:true, code:0x0} (runtime error: index out of range [0] with length 0)
goroutine 1620 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1a15ce0, 0xc0012ca180)
        /Users/xcoulon/code/go/pkg/mod/k8s.io/apimachinery@v0.18.3/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /Users/xcoulon/code/go/pkg/mod/k8s.io/apimachinery@v0.18.3/pkg/util/runtime/runtime.go:48 +0x82
panic(0x1a15ce0, 0xc0012ca180)
        /usr/local/Cellar/go@1.14/1.14.7/libexec/src/runtime/panic.go:969 +0x166
github.com/codeready-toolchain/host-operator/controllers/toolchainstatus.(*Reconciler).membersHandleStatus(0xc000080dc0, 0x1e05d60, 0xc0008cfa60, 0xc00076ba20, 0x1)
        /Users/xcoulon/code/go/src/github.com/codeready-toolchain/host-operator/controllers/toolchainstatus/toolchainstatus_controller.go:354 +0xb06
g
```

### CODE REVIEW

File: controllers/toolchainstatus/toolchainstatus_controller.go [hunk 1]
```diff
@@ -351,7 +351,11 @@ func (r *Reconciler) membersHandleStatus(logger logr.Logger, toolchainStatus *to
 			logger.Error(fmt.Errorf("member cluster %s not ready", memberCluster.Name), "the memberstatus ready condition is not true")
 			ready = false
 		}
-		r.Log.Info("adding member status", "member_name", memberCluster.Name, string(memberStatus.Conditions[0].Type), memberStatus.Conditions[0].Status)
+		if readyCond, found := condition.FindConditionByType(memberStatusObj.Status.Conditions, toolchainv1alpha1.ConditionReady); found {
```
Comment (xcoulon) [inline controllers/toolchainstatus/toolchainstatus_controller.go:354]: wouldn't it be possible to merge this new if/else statement with the one above on line 349?

Comment (MatousJobanek) [reply controllers/toolchainstatus/toolchainstatus_controller.go:354]: yeah, technically yes

File: controllers/toolchainstatus/toolchainstatus_controller.go [hunk 2]
```diff
@@ -351,7 +351,11 @@ func (r *Reconciler) membersHandleStatus(logger logr.Logger, toolchainStatus *to
...
+			r.Log.Info("adding member status", "member_name", memberCluster.Name, string(readyCond.Type), readyCond.Status)
+		} else {
+			r.Log.Info("adding member status", "member_name", memberCluster.Name, "Ready condition", "empty")
```
Comment (alexeykazakov) [inline controllers/toolchainstatus/toolchainstatus_controller.go:357]: Missing condition actually means Unknown.

**Review by alexeykazakov — APPROVED** [review]
Good catch

File: controllers/toolchainstatus/toolchainstatus_controller.go (hunk 1 above)
Comment (MatousJobanek) [reply controllers/toolchainstatus/toolchainstatus_controller.go]: I'm not sure if the code is better in this way: c459cca

File: controllers/toolchainstatus/toolchainstatus_controller.go (hunk 2 above)
Comment (MatousJobanek) [reply controllers/toolchainstatus/toolchainstatus_controller.go]: fixed: c459cca

**Pull request merged**